vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js

```diff
@@ -0,0 +1,1681 @@
+        }
+        el.off(ev, ch);
+      }
+    },
+    _buildEvents: function(){
+      if (this.isInput){ // single input
+        this._events = [
+          [this.element, {
+            focus: $.proxy(this.show, this),
+            keyup: $.proxy(function(e){
+              if ($.inArray(e.keyCode, [27,37,39,38,40,32,13,9]) === -1)
+                this.update();
+            }, this),
+            keydown: $.proxy(this.keydown, this)
+          }]
+        ];
+      }
+      else if (this.component && this.hasInput){ // component: input + button
+        this._events = [
+          // For components that are not readonly, allow keyboard nav
+          [this.element.find('input'), {
+            focus: $.proxy(this.show, this),
+            keyup: $.proxy(function(e){
+              if ($.inArray(e.keyCode, [27,37,39,38,40,32,13,9]) === -1)
+                this.update();
+            }, this),
+            keydown: $.proxy(this.keydown, this)
+          }],
+          [this.component, {
+            click: $.proxy(this.show, this)
+          }]
+        ];
+      }
+      else if (this.element.is('div')){  // inline datepicker
+        this.isInline = true;
+      }
+      else {
+        this._events = [
+          [this.element, {
+            click: $.proxy(this.show, this)
+          }]
+        ];
+      }
+      this._events.push(
+        // Component: listen for blur on element descendants
+        [this.element, '*', {
+          blur: $.proxy(function(e){
+            this._focused_from = e.target;
+          }, this)
+        }],
+        // Input: listen for blur on element
+        [this.element, {
+          blur: $.proxy(function(e){
+            this._focused_from = e.target;
+          }, this)
+        }]
+      );
+
+      this._secondaryEvents = [
+        [this.picker, {
+          click: $.proxy(this.click, this)
+        }],
+        [$(window), {
+          resize: $.proxy(this.place, this)
+        }],
+        [$(document), {
+          'mousedown touchstart': $.proxy(function(e){
+            // Clicked outside the datepicker, hide it
+            if (!(
+              this.element.is(e.target) ||
+              this.element.find(e.target).length ||
+              this.picker.is(e.target) ||
+              this.picker.find(e.target).length
+            )){
+              this.hide();
+            }
+          }, this)
+        }]
+      ];
+    },
+    _attachEvents: function(){
+      this._detachEvents();
+      this._applyEvents(this._events);
+    },
+    _detachEvents: function(){
+      this._unapplyEvents(this._events);
+    },
+    _attachSecondaryEvents: function(){
+      this._detachSecondaryEvents();
+      this._applyEvents(this._secondaryEvents);
+    },
+    _detachSecondaryEvents: function(){
+      this._unapplyEvents(this._secondaryEvents);
+    },
+    _trigger: function(event, altdate){
+      var date = altdate || this.dates.get(-1),
+        local_date = this._utc_to_local(date);
```

Comment:
Identifier 'local_date' is not in camel case.<br>Identifier '_utc_to_local' is not in camel case.